crates/db-address-space/src/address_space.rs

```diff
@@ -0,0 +1,979 @@
+// Copyright (C) 2021 Alibaba Cloud. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types for address space information.
+
+use std::ffi::CString;
+use std::fs::{File, OpenOptions};
+use std::io::{Read, Write};
+use std::os::unix::io::FromRawFd;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
+use nix::sys::memfd;
+use vm_memory::{
+    Address, AtomicAccess, Bytes, FileOffset, GuestAddress, GuestMemoryError, GuestMemoryMmap,
+    GuestMemoryRegion, GuestUsize, MemoryRegionAddress,
+};
+
+/// Internal types for Address space.
+#[cfg(not(feature = "atomic-guest-memory"))]
```

Comment:
Could you add some description for atomic-guest-memory since it's hard for people to understand its usage

Reply:
I've added some description pls have a look if it is clear